File: Gradle/src/main/java/com/pqixing/modularization/manager/GitProject.java
package com.pqixing.modularization.manager;

import org.eclipse.jgit.api.Git;

/**
 * Created by pqixing on 17-12-7.
 */

public class GitProject {
    /**
     * git工程名称,指目录名称
     */
    public String name;

    /**
     * git地址
     */
    public String gitUrl;

    /**
     *
     */
    public String introduce;
    /**
     * 根目录的名称
     */
    public String rootName;

    public GitInfo gitInfo;


    public GitProject(String name, String gitUrl, String introduce, String rootName) {
        this.name = name;
        this.gitUrl = gitUrl;
        this.introduce = introduce;
        this.rootName = rootName;
    }

    public GitProject() {
    }

    public void loadGitInfo(Git git) {

    }

    public static class GitInfo {

    }
}

File: Gradle/src/main/java/com/pqixing/modularization/manager/GitProject.kt
package com.pqixing.modularization.manager

import org.eclipse.jgit.api.Git

/**
 * Created by pqixing on 17-12-7.
 */

class GitProject {
    /**
     * git工程名称,指目录名称
     */
    var name: String = ""

    /**
     * git地址
     */
    var gitUrl: String = ""

    /**
     *
     */
    var introduce: String = ""
    /**
     * 根目录的名称
     */
    var rootName: String = ""

    var gitInfo: GitInfo? = null


    constructor(name: String, gitUrl: String, introduce: String, rootName: String) {
        this.name = name
        this.gitUrl = gitUrl
        this.introduce = introduce
        this.rootName = rootName
    }

    constructor() {}

    fun loadGitInfo(git: Git) {

    }

}

/**
 * git相关信息
 */
data class GitInfo(var commit: String = ""
                   , var author: String = ""
                   , var lastTime: String = ""
                   , var message: String = ""
                   , var branch: String = "") {
    var log: List<String>? = null
}
